### PR TITLE
Modify regex_revalidate so that traffic_ctl may be used to trigger updates

### PR DESCRIPTION
**Pull request by jrushford**
@zwoop Would like to back port this change to 7.1.x.  We are moving to 7.1 and need this change.

(cherry picked from commit 4da3d7c2408be0ff5424ad8ffd518b9cd2fca174)

### CODE REVIEW

**Comment by jrushford** [comment]
[approve ci]

File: lib/ts/Ptr.h [hunk 1]
```diff
@@ -111,10 +111,8 @@ template <class T> class Ptr
 
   T *operator->() const { return (m_ptr); }
   T &operator*() const { return (*m_ptr); }
-
   // Making this explicit avoids unwanted conversions.  See https://en.wikibooks.org/wiki/More_C%2B%2B_Idioms/Safe_bool .
   explicit operator bool() const { return m_ptr != nullptr; }
-
   int
   operator==(const T *p)
   {
```
Comment (jrushford) [inline lib/ts/Ptr.h]: This change is not part of this PR nor was it part of the original diff.  This change was made to specificall fix a clang format problem that must have existed on 7.1.x.

**Comment by jrushford** [comment]
[approve ci autest]

**Comment by jrushford** [comment]
[approve ci]

**Comment by jrushford** [comment]
[approve ci]

**Comment by jrushford** [comment]
[approve ci]

**Comment by jrushford** [comment]
The autest failure is not due to any changes with this PR.  @dragon512 tells me that he needs to backport some autest fixes to the 7.1.x branch in ci.

**Comment by jrushford** [comment]
[approve ci]

**Comment by zwoop** [comment]
@jrushford I'd like to defer this to 7.1.3, unless there are very strong reasons for this. Since it's a "feature", I'd be hard pressed to respin 7.1.2 again. :)

**Comment by jrushford** [comment]
@zwoop that is fine, go ahead and defer.  We’re making ready to deploy 7.1.2 to production and we need this but, we can cherry pick it.

**Comment by jrushford** [comment]
[approve ci]

**Comment by zwoop** [comment]
[approve ci autest]
